### PR TITLE
Add commonjs2 library build

### PR DESCRIPTION
**Pull request by derekstavis**
Fix issue #30  by adding a build step and distributing compiled files.

Webpack configuration is being inherited from `react-scripts` production config and tweaked to achieve a proper commonjs2 library build, featuring:

* Minified and unminified `.js` targets
* Source maps for both targets

By default, `package.json` have `main` pointing to unminified build. Users wishing to include the minified version would need to use `require('react-event-components/dist/react-event-components.min.js')`.

### CODE REVIEW

**Comment by derekstavis** [comment]
Please give some testing to the libraries compiled in `dist` directory.

**cc** @felquis @MarcoWorms @grvcoelho

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  library: 'react-event-components',
+  libraryTarget: 'commonjs2',
+}
+
+config.externals = {
+  'react': 'react',
+  'react-dom': 'react-dom',
+}
+
+const blacklist = [
```
Comment (grvcoelho) [inline webpack.config.js]: Oh, nevermind, I got it! 👍

File: webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+process.env.NODE_ENV = 'production'
+
+const path = require('path')
+
+const UglifyJsPlugin = require('webpack').optimize.UglifyJsPlugin
+const HtmlWebpackPlugin = require('html-webpack-plugin')
+const ManifestPlugin = require('webpack-manifest-plugin')
+const InterpolateHtmlPlugin = require('react-dev-utils/InterpolateHtmlPlugin')
+
+const config = module.exports = require('react-scripts/config/webpack.config.prod.js')
```
Comment (MarcoWorms) [inline webpack.config.js]: why not make this clearer with 2 statements?

**Comment by derekstavis** [comment]
I have worked the reviewed aspects and also added comments to the whole file to ease understanding of what is done and why it is done this way. 👍

File: webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  HtmlWebpackPlugin,
+  ManifestPlugin,
+  InterpolateHtmlPlugin,
+].map(c => c.name)
+
+// Reduce the original plugin list in order to create
+// a new list of plugins without blacklisted ones.
+
+config.plugins = config.plugins.reduce((acc, item) => {
+  if (blacklist.indexOf(item.constructor.name) !== -1) {
```
Comment (MarcoWorms) [inline webpack.config.js]: @felquis

File: package.json [hunk 4]
```diff
@@ -4,20 +4,22 @@
...
-  "main": "src/react-event-components/index.js",
+  "main": "dist/react-event-components.js",
   "devDependencies": {
     "gh-pages": "0.11.0",
     "react-scripts": "0.7.0"
   },
   "dependencies": {
     "react": "^15.3.2",
-    "react-dom": "^15.3.2"
+    "react-dom": "^15.3.2",
```
Comment (felquis) [inline package.json]: ?

Comment (felquis) [inline package.json]: We should put `react-dom` and `react` in `devDependencies`

File: webpack.config.js (hunk 3 above)
Comment (felquis) [inline webpack.config.js]: Oh

Comment (felquis) [inline webpack.config.js]: node 6.7

```
let list = [1,2,3,4]

list.includes(2) // true
```

**Comment by felquis** [comment]
https://github.com/pagarme/react-event-components/pull/31#issuecomment-263174947 @derekstavis you rock! 

You deserve 🍬💐🍬💐  for the comments 😄 

GIF of actor Dwayne Johnson acting as The Rock smiling!

File: webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+// Use the imported modules to create a blacklist of
+// all modules constructor names we want to squash
+// from original (react-scripts) config.
+
+const blacklist = [
+  HtmlWebpackPlugin,
+  ManifestPlugin,
+  InterpolateHtmlPlugin,
+].map(c => c.name)
```
Comment (felquis) [inline webpack.config.js]: What is `c`, a car name? Or customers, channel

File: webpack.config.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  HtmlWebpackPlugin,
+  ManifestPlugin,
+  InterpolateHtmlPlugin,
+].map(plugin => plugin.name)
+
+// Reduce the original plugin list in order to create
+// a new list of plugins without blacklisted ones.
+
+config.plugins = config.plugins.reduce((acc, plugin) => {
+  if (blacklist.includes(plugin.constructor.name)) {
```
Comment (felquis) [inline webpack.config.js]: ❤️

**Pull request merged**